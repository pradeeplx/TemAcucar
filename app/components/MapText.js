import React from 'react'
import { View, Image, Platform } from 'react-native'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import Sentence from "./Sentence"

export default MapText = ({ children }) => (
  <View style={{
    position: 'absolute',
    bottom: 6,
    left: 0,
    right: 0,
    alignItems: 'center',
  }}>
    <View style={{
      backgroundColor: Colors.white,
      paddingVertical: 3,
      paddingHorizontal: 30,
      borderRadius: 12,
    }}>
      <Sentence style={[{
        color: Colors.blue,
        textAlign: 'center',
        fontSize: 10 * fontFactor(),
      }, (Platform.OS === 'ios' ? { fontFamily: 'Avenir-Black' } : { fontWeight: 'bold' })]}>
        { children }
      </Sentence>
    </View>
  </View>
)
