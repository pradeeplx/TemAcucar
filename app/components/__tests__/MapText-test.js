import 'react-native'
import React from 'react'
import MapText from '../MapText'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <MapText>Foo</MapText>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
