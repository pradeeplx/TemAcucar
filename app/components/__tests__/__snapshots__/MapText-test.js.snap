exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 6,
      "left": 0,
      "position": "absolute",
      "right": 0,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "paddingHorizontal": 30,
        "paddingVertical": 3,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#44B5B1",
              "fontSize": 15,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
          ],
        ]
      }>
      Foo
    </Text>
  </View>
</View>
`;
